lib/logmsg/nvtable.c

```diff
@@ -187,6 +187,8 @@ nv_table_resolve_indirect(NVTable *self, NVEntry *entry, gssize *length)
    * non-zero terminated strings properly handled, thus the caller has
    * to supply a non-NULL value_len */
 
+  g_assert(length != NULL);
```

Comment:
I would allow length to be null, in that case we just simply not write to the output parameter, similarly as we do in line 181.
For example in the python code, we indeed do not need the length, we could avoid the dummy parameter.

Reply:
@furiel : based on the comment it is better to assert, IMHO

Reply:
Hmmm that's a problem because it would mean PyBytes_FromString() could be possibly called on a non-null terminated string. I'm afraid we should strndup with value_len on the caller site then.

Reply:
log_msg_get_value() returning a length field is important to support both:

1) indirect values
2) embedded NULs

the 2nd is not yet reality and I am not sure it would ever be, but the first one is.

Please note however that builtin values (e.g. LM_V_MESSAGE and friends) are always nul terminated and never indirect. (there's an assert to ensure that in log_msg_set_value_indirect)

Reply:
With this command:

```
$ git grep log_msg_get_value\( | grep -v '&' | grep -v test | grep -v LM_V
lib/logmsg/logmsg.c:__log_msg_get_value(const LogMessage *self, NVHandle handle, gssize *value_len)
lib/logmsg/logmsg.h:log_msg_get_value(const LogMessage *self, NVHandle handle, gssize *value_len)
lib/logmsg/logmsg.h:  return log_msg_get_value(self, handle, value_len);
modules/python/python-logmsg.c:  const gchar *value = log_msg_get_value(py_msg->msg, handle, NULL);
```

I've doublechecked that the only call-site that was doing this was indeed the Python implementation (well, except for tests).

Reply:
one more comment: there's a macro to ensure a string is NUL terminated, just for this purpose, it is called APPEND_ZERO in str-utils.h

What it does that it checks if there's a NUL at the end of the string, and if there's none, it copies the string to an alloca() provided buffer. It is much faster than doing this with all the values, if we except that most of them will be NUL terminated.

Reply:
done